crates/astria-celestia-jsonrpc-client/src/rpc_impl/blob.rs

```diff
@@ -0,0 +1,132 @@
+//! The Celestia JSON RPC blob API.
+//!
+//! Many of the constants are taken from these two sources:
+//!
+//! + [celestia-app:965eaf global_consts.go](https://github.com/celestiaorg/celestia-app/blob/965eafb4357376aec31f84f3628f7703c5587f9a/pkg/appconsts/global_consts.go)
+//! + [celestia-app:965eaf namespace/consts.go](https://github.com/celestiaorg/celestia-app/blob/965eafb4357376aec31f84f3628f7703c5587f9a/pkg/namespace/consts.go)
+use jsonrpsee::proc_macros::rpc;
+use serde::{
+    Deserialize,
+    Serialize,
+};
+
+use crate::serde::Base64Standard;
+
+/// The full theoretical length of the celestia namespace ID in bytes.
+///
+/// In practice the actual length available (at least for version 0 of the namespace
+/// API) is [`NAMESPACE_ID_ACTUAL_LEN`] (currently 10 bytes).
+///
+/// From [celestia-app:965eaf global_consts.go#L22](https://github.com/celestiaorg/celestia-app/blob/965eafb4357376aec31f84f3628f7703c5587f9a/pkg/appconsts/global_consts.go#L22)
+const NAMESPACE_ID_LEN: usize = 28;
+
+/// The length of the celestia namespace version in bytes.
+///
+/// From [celestia-app:965eaf global_consts.go#L16](https://github.com/celestiaorg/celestia-app/blob/965eafb4357376aec31f84f3628f7703c5587f9a/pkg/appconsts/global_consts.go#L16)
+const NAMESPACE_VERSION_LEN: usize = 1;
+
+/// The total length of the celestia namespace byte slice as ingested by their API.
+///
+/// Currently defined as the length of the namespace ID and the length of the namespace version.
+/// From [celestia-app:965eaf global_consts.go#L25](https://github.com/celestiaorg/celestia-app/blob/965eafb4357376aec31f84f3628f7703c5587f9a/pkg/appconsts/global_consts.go#L25)
+const NAMESPACE_LEN: usize = NAMESPACE_ID_LEN + NAMESPACE_VERSION_LEN;
+
+/// The number of zeros that the namespace ID is prefixed withe for version 0 of the namespace API.
+///
+/// From [celestia-app:965eaf namespace/consts.go#L28](https://github.com/celestiaorg/celestia-app/blob/965eafb4357376aec31f84f3628f7703c5587f9a/pkg/namespace/consts.go#L28)
+pub(crate) const NAMESPACE_VERSION_ZERO_PREFIX_LEN: usize = 18;
+
+/// The actual number of bytes that is available to the user to construct a namespace.
+///
+/// From [celestia-app:965eaf namespace/consts.go#L32](https://github.com/celestiaorg/celestia-app/blob/965eafb4357376aec31f84f3628f7703c5587f9a/pkg/namespace/consts.go#L32)
+pub const NAMESPACE_ID_AVAILABLE_LEN: usize = NAMESPACE_ID_LEN - NAMESPACE_VERSION_ZERO_PREFIX_LEN;
+
+/// The commitment of a blob.
+///
+/// At this time it is not clear how this is constructed, so should probably be left empty.
```

Comment:
this is probably hash of the blob data or something like that, will check

Reply:
I will make an issue to investigate this so we don't lose track.